org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/configuration/CompositeConfiguration.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Gradle Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+
+package org.eclipse.buildship.core.internal.configuration;
+
+import java.io.File;
+
+import org.eclipse.core.runtime.IAdaptable;
+
+/**
+ * Configuration for a workspace composite in a Gradle build.
+ *
+ * @author Sebastian Kuzniarz
+ */
+public interface CompositeConfiguration {
+
+    File getCompositeDir();
+
+    IAdaptable[] getProjectList();
+
+    BuildConfiguration getBuildConfiguration();
+
+    Boolean projectAsCompositeRoot();
+
+    File getRootProject();
```

Comment:
⭕ getRootProject and getCompositeDir are not the same?

Reply:
getCompositeDir returns the folder of the composite preference file where getRootProject returns the custom project root but right now I'm reworking the CompositeConfiguration class, because I think the path doesn't have to be stored here

Reply:
Ok, let me know if you need more input.

Reply:
I changed it to `String getCompositeName()`, so every CompositeConfiguration consists of a composite name, included builds, a build configuration, and an optional composite root project, which makes a lot more sense I guess

Reply:
Sounds better :)
How does the optional composite root project compare to the root project location in the build configuration? Is the composite root even needed?

Reply:
Actually, it's the same thing, you're right! I'll keep the projectAsCompositeRoot() function, so the configuration still contains information whether the root project is synthetic or a real one